packages/loot-core/src/server/accounts/sync.ts

```diff
@@ -645,6 +655,16 @@ export async function reconcileTransactions(acctId, transactions) {
           acctId,
         ],
       );
+
+      // Sort the matched transactions according to the distance from the original
+      // transactions date. i.e. if the original transaction is in 21-02-2024 and
+      // the matched transactions are: 20-02-2024, 21-02-2024, 29-02-2024 then
+      // the resulting data-set should be: 21-02-2024, 20-02-2024, 29-02-2024.
+      fuzzyDataset = fuzzyDataset.sort((a, b) => {
+        const aDistance = Math.abs(db.toDateRepr(trans.date) - a.date);
+        const bDistance = Math.abs(db.toDateRepr(trans.date) - b.date);
+        return aDistance > bDistance ? 1 : -1;
+      });
```

Comment:
It would be nice to unify `reconcileExternalTransactions` and `reconcileTransactions`. But that's a story for another day.